### ISSUE TITLE
Build fails on Windows - make-locales

### ISSUE THREAD

**Issue by hpmachining** [Pomax/BezierInfo-2#112]
This may turn out to be a support request instead of an issue, but I can't get the project to build on either Windows 7 or Windows 10. I get the following error after `node tools/make-locales` during the build process: 

> FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory

I tried increasing the heap size with `node --max-old-space-size=8196 tools/make-locales` but that just made it run for over an hour before getting the error. I should note that the project builds for me without problems in Linux.  I have tried nodejs version 6.10.2 / npm version 3.10.10 and nodejs version 7.7.3 / npm version 4.5.0. Any thoughts on what the problem is?

Here is some basic info on the two computers:
Laptop
Intel Core i5-3210M CPU, 2.50 GHz.
6 GB RAM
Dual boot Windows 10 / Arch Linux

Desktop
Intel Core i7-3770K, 3.50 GHz
32 GB RAM
Windows 7

**Comment by Pomax** [comment]
I've run this on several windows 10  with Node 7 without any special settings.

**Comment by hpmachining** [comment]
No, I am trying to build "as is". To be sure, I just tried again with a fresh clone of your repository on both my Windows 7 desktop and my Arch Linux laptop with the same results, Linux works, Windows doesn't. Some additional notes, when I run `npm install` I get some warnings on both systems. I get 2 about skipping optional dependency for fsevents and then the following 2:

>npm WARN react-component-visibility@0.0.8 requires a peer of react@^0.13 but none was installed.
>npm WARN eslint-plugin-react@6.10.3 requires a peer of eslint@^2.0.0 || ^3.0.0 but none was installed.

So I tried changing the package.json entries to `"eslint": "^3.0.0"` and `"react-component-visibility": "^2.0.0"`, which gets rid of the last 2 warnings but doesn't help the build. This was after first trying to build "as is".

Since you are using Node 7 without issue I will go back to that version and try again but I don't think that is the problem since that is the version I tried first. I am at a loss as to what to try next. I guess I can try setting up a shell with a stripped down path and environment to eliminate the possibility that something else loaded on my computers is interfering somehow, but that doesn't seem likely to me. I am open to suggestions if you have any. Thank you!

**Comment by Pomax** [comment]
seriously: add some console logs to the functions in https://github.com/Pomax/BezierInfo-2/blob/master/tools/make-locales.js and https://github.com/Pomax/BezierInfo-2/blob/master/tools/lib/chunk.js, so that every function starts with:
```
function whatevername(whatever, args, etc) {
  console.log("entering whatevername");
  ...
}
```
and see what the console spits out before it hits the memory limit. I've never seen this happen and I run the compile on three different machines quite frequently so I'd be very interested in seeing that log.

**Comment by hpmachining** [comment]
I used your method to track it down to line 50 `data = chunk(data);` inside the `processLocation`. I added console logs to `chunk.js`, but it doesn't stop on the same function every time. It does seem to pause a lot on `fixMarkDownLinks` while it's running. It's getting late, so I'll try to look at it some more tomorrow.

**Comment by Pomax** [comment]
I just did a full node_modules wipe (with npm cache clean and npm prune to make sure there's no hidden secret bits left), and then ran a new npm install + npm run build, but still no heap errors or getting stuck in make-locales. Looking at npm's memory use while the build runs, it never exceeds 425MB.

**Comment by hpmachining** [comment]
I am seeing similar memory usage when building in Linux. It's just so strange that I am having the same issue on 2 different computers with 2 different versions of Windows. I have uninstalled Node / npm, deleted all traces of them, ran ccleaner and am in the process of doing a full system virus scan on both computers. When that completes I will try again from scratch. Obviously with 73 forks and nobody else having an issue there is something going on on my end. I'll keep at it and let you know if I get it working. If you have any suggestions I would be glad to hear them, but I already feel like I have taken up too much of your time. Thank you!

**Comment by Pomax** [comment]
no problem on my end - I'm always up for helping think of possible causes
